### ISSUE TITLE
Cannot use background image in master slide

### ISSUE THREAD

**Issue by blu3print** [PHPOffice/PHPPresentation#330]
I currently am using the same code over and over to create the slide look and feel.

It includes using a background image. When I use the same code with createMasterSlide() it says the image cannot be displayed in the master section.

I tried marking the image as a placeholder but that does not seem to exist.

What is the correct way to do this?

**Comment by Progi1984** [comment]
@blu3print Could you give us a sample code for testing ?

**Comment by blu3print** [comment]
@Progi1984 

I can't share all of my code but here is what I had tried previously, relevant to the master slide generation:

`//General global objects
    $oBkgImageTitle = new Image();
    $oBkgImageTitle->setPath('wwwroot/assets/img/template-bg.jpg');
    $oBkgImageSlide = new Image();
    $oBkgImageSlide->setPath('wwwroot/assets/img/template-bg-2017.png');
//Master slides
$masterslides[$current_masterslide] = $objPHPPresentation->createMasterSlide();
    $masterslides[$current_masterslide]->setBackground($oBkgImageSlide);

    //Title of slide
    $richtext = $masterslides[$current_masterslide]->createRichTextShape()
            ->setHeight(30)
            ->setWidth(1000)
            ->setOffsetX(45)
            ->setOffsetY(25);
    $textRun = $richtext->createTextRun('Slide Title');
    $textRun->getFont()->setSize(22)
                        ->setItalic(true)
                        ->setColor(new StyleColor('FF000000'));
`

This ends up "generating" the master slides but what should be the background image (which works on normal slide generation) is a big red X accompanied by a "corrupt" error when loading the powerpoint. 

I generate all of my powerpoint from this package so I do not have a "base" powerpoint that I am reading and manipulating that has the background image baked in already.

The users are just complaining about not being able to hit new slide and have the background the other slides have already be there so I thought the master slide mechanism would accomplish that.

I hope that is helpful!

**Comment by forexgg** [comment]
I have the same issue, only draw shape is working in master slide, not images.

**Comment by Progi1984** [comment]
@forexgg @blu3print Could you test the develop branch ?

**Comment by forexgg** [comment]
sure, I’ll test it later.

**Comment by forexgg** [comment]
Yeah, master slides shows the background image fine now.



      But trying to load the existing PPTX, got error below: 



      imagecreatefromstring(): gd-png: fatal libpng error: IDAT: invalid literal/length code
     
      in PowerPoint2007.php (line 731)

**Comment by Progi1984** [comment]
@forexgg Have you got the problem with samples ?

If it isn't the case, could you give us your existing PPTX ?

**Comment by forexgg** [comment]
Hi Progi,

Sure, I attached one of our program to this email. This file has 3 master slides and 4 layouts.

I tried not use existing PPTX and generate slides from scratch. I can generate new master slide. But there are not much documents that I can reference to generate new layouts for particular master slide. And also, apply different new slides to different layouts. If you have part of the example code, that would help a lot.

So, I tried to load existing slides and modify it, but it won’t load.

Thank you for your kindly help.

Steven
